vm/src/object/core.rs

```diff
@@ -641,8 +642,15 @@ impl PyObject {
     }
 
     #[inline(always)]
-    pub(crate) fn class_lock(&self) -> &PyRwLock<PyTypeRef> {
-        &self.0.typ
+    pub fn class(&self) -> &Py<PyType> {
+        self.0.typ.deref()
+    }
+
+    pub fn set_class(&self, typ: PyTypeRef, vm: &VirtualMachine) {
+        let old = unsafe { self.0.typ.swap(typ) };
+        if let Some(frame) = vm.current_frame() {
+            frame.temperary_refs.lock().push(old.into());
+        }
```

Comment:
If the swapped object is always expected to put in temporary refs,(I guess so)
frame or atomicref is better to have a safe interface to do the job.